core/chains/evm/gas/rollups/zkSync_l1_oracle.go

```diff
@@ -0,0 +1,254 @@
+	SystemContextAddress = "0x000000000000000000000000000000000000800B"
+
+	// ZksyncGasInfo_GetL2GasPerPubDataBytes is the a hex encoded call to:
+	// function gasPerPubdataByte() external view returns (uint256 gasPerPubdataByte);
+	SystemContext_gasPerPubdataByteMethod = "gasPerPubdataByte"
+	ZksyncGasInfo_getGasPerPubdataByteL2  = "0x7cb9357e"
+
+	// ZksyncGasInfo_GetL2GasPrice is the a hex encoded call to:
+	// `function gasPrice() external view returns (uint256);`
+	SystemContext_gasPriceMethod = "gasPrice"
+	ZksyncGasInfo_getGasPriceL2  = "0xfe173b97"
+)
+
+func NewZkSyncL1GasOracle(lggr logger.Logger, ethClient l1OracleClient) *zkSyncL1Oracle {
+	var systemContextAddress, gasPerPubdataMethod, gasPerPubdataSelector, l2GasPriceMethod, l2GasPriceSelector string
+
+	systemContextAddress = SystemContextAddress
+	gasPerPubdataMethod = SystemContext_gasPerPubdataByteMethod
+	gasPerPubdataSelector = ZksyncGasInfo_getGasPerPubdataByteL2
+
+	l2GasPriceMethod = SystemContext_gasPriceMethod
+	l2GasPriceSelector = ZksyncGasInfo_getGasPriceL2
+
+	return &zkSyncL1Oracle{
+		client:     ethClient,
+		pollPeriod: PollPeriod,
+		logger:     logger.Sugared(logger.Named(lggr, "L1GasOracle(zkSync)")),
+		chainType:  chaintype.ChainZkSync,
+
+		systemContextAddress:  systemContextAddress,
+		gasPerPubdataMethod:   gasPerPubdataMethod,
+		gasPerPubdataSelector: gasPerPubdataSelector,
+		l2GasPriceMethod:      l2GasPriceMethod,
+		l2GasPriceSelector:    l2GasPriceSelector,
+
+		chInitialised: make(chan struct{}),
+		chStop:        make(chan struct{}),
+		chDone:        make(chan struct{}),
+	}
+}
+
+func (o *zkSyncL1Oracle) Name() string {
+	return o.logger.Name()
+}
+
+func (o *zkSyncL1Oracle) Start(ctx context.Context) error {
+	return o.StartOnce(o.Name(), func() error {
+		go o.run()
+		<-o.chInitialised
+		return nil
+	})
+}
+func (o *zkSyncL1Oracle) Close() error {
+	return o.StopOnce(o.Name(), func() error {
+		close(o.chStop)
+		<-o.chDone
+		return nil
+	})
+}
+
+func (o *zkSyncL1Oracle) HealthReport() map[string]error {
+	return map[string]error{o.Name(): o.Healthy()}
+}
+
+func (o *zkSyncL1Oracle) run() {
+	defer close(o.chDone)
+
+	t := o.refresh()
+	close(o.chInitialised)
+
+	for {
+		select {
+		case <-o.chStop:
+			return
+		case <-t.C:
+			t = o.refresh()
+		}
+	}
+}
+func (o *zkSyncL1Oracle) refresh() (t *time.Timer) {
+	t, err := o.refreshWithError()
+	if err != nil {
+		o.SvcErrBuffer.Append(err)
+	}
+	return
+}
+
+func (o *zkSyncL1Oracle) refreshWithError() (t *time.Timer, err error) {
+	t = time.NewTimer(utils.WithJitter(o.pollPeriod))
+
+	ctx, cancel := o.chStop.CtxCancel(evmclient.ContextWithDefaultTimeout())
+	defer cancel()
+
+	price, err := o.CalculateL1GasPrice(ctx)
+	if err != nil {
+		return t, err
+	}
+
+	o.l1GasPriceMu.Lock()
+	defer o.l1GasPriceMu.Unlock()
+	o.l1GasPrice = priceEntry{price: assets.NewWei(price), timestamp: time.Now()}
+	return
+}
+
+// For zkSync l1_price_per_pubdata ~= gas_price_on_l2 * l2_gas_PerPubdataByte
```

Comment:
nit: explicitly specify is it per byte, like `l1_price_per_pubdata_byte`, as opposed to pubdata gas or slot